pennylane/math/quantum.py

```diff
@@ -972,6 +975,118 @@ def sqrt_matrix(density_matrix):
     return vecs @ qml.math.diag(qml.math.sqrt(evs)) @ qml.math.conj(qml.math.transpose(vecs))
 
 
+def sqrt_matrix_sparse(density_matrix):

```

Comment:
Based on the conversation in the other thread, can we rename this ? Also, anywhere we use DM throughout too?

Reply:
@mlxd How about `sparse_matrix` here?

Reply:
I don't see other places in this PR that uses DM

Reply:
Yep, that's fine. The docstring can explain the intention and valid structure.